.github/workflows/stub-tests.yml

```diff
@@ -0,0 +1,45 @@
+name: Stub tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Go ${{ matrix.go-version }} - ${{ matrix.os }}/${{ matrix.test-arch }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.18.x]
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        test-arch: [amd64]
+        race: ['-race']
+
+        include:
+          - go-version: 1.18.x
+            test-arch: '386'
+            os: ubuntu-latest
+            race: ''
+          - go-version: 1.18.x
+            test-arch: '386'
+            os: windows-latest
+            race: ''
+
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - uses: actions/setup-go@268d8c0ca0432bb2cf416faae41297df9d262d7f
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Run tests on stubs
+        shell: bash
+        env:
+          GOARCH: ${{ matrix.test-arch }}
+        run: |
+          ./bin/test-without-stubs test-stubs
```

Comment:
That the "test-without-stubs" script now also runs tests with stubs is very confusing. The script should be renamed.

Honestly, I would prefer if you could create a separate PR for adding this bit. It makes this PR even larger, it introduces new problems as you wrote and all of that delays that we can get the other good things you added here merged.

If you extract that bit into a new PR, make sure this PR is up to date with master, then we can resolve whatever issues are left regarding to the original goal (lint and run tests for generator and gomod-sync) and then we can merge this. More can always be done later. Please keep this in mind in general, small PRs are always better, always try to avoid increasing the scope.

Reply:
> Honestly, I would prefer if you could create a separate PR for adding this bit
What should be part of the other PR? Renaming `test-without-stubs`?

Reply:
I meant the whole "running the tests against the stubs and check they fail" part. If you can get it working properly, then you can rename the file and keep it in. Otherwise, as mentioned above, it would be better to tackle "running tests against stubs" in a separate PR.

Reply:
I renamed the script, is the name fine like this?

I have added a code passing the tests for Gigasecond and as you can see, CI fails, so it is working properly

Reply:
Cool, what did you change compared to before where you said it is not working?

The name is fine now.

Reply:
I did not change anything, I only adjusted the stub to be really passing tests (must be reverted before merging). The problem from https://github.com/exercism/go/pull/2451 is not detected, because in CI the tests do not pass:https://github.com/exercism/go/actions/runs/3073998679/jobs/4966490822#step:4:1840